### DISCUSSION TITLE
CNC problems with noise/alarms

### DISCUSSION THREAD

**Discussion by spannerman52** [winder/Universal-G-Code-Sender#1615] [Help]
Can anyone help please I am new to CNC I've just built my first from scratch and used Nema 23 motors with microswitches as homing switches. The router I have is a cheap Chinese ER11 with a remote power supply which I have mounted alongside my CNC power supply in a box. I can jog the machine until I turn on the router motor then after a couple of jog movements everything stops (apart from the router. The only way I can get it to move is switch off the router and disconnect UGS re-connect and home cycle. What have I done wrong?  All my switches and motors are wired with screened cable the router isn't. I was thinking of buying a Makita trim router to mount on the machine but I'd like to solve this problem first. Is it noise and if so how do I find out where and fix it. I've watched so many videos about setting debounce (whatever that is?) and fitting optocouplers but it's all getting a bit exasperating 😩 I hope someone can help me? Thanks

**Comment by breiler** [comment]
I've battled similar problems, but yours sounds more serious. In my case the controller got ghost alarms due to EMI triggering the limit switches, but the controller was still active.

Is it possible to replace the spindle power cable? 
In my case the EMI was so bad that I could feel a "tingle" when touching the chassis of the machine when the spindle was on. I bought a shielded 220v three phase cable with a separate ground and everything got more stable and the "tingle" was gone. It was quite expensive, but I suspect that this is needed in your case to prevent EMI from your spindle to resetting your controller.

I did however occasionally get ghost signals triggering the limit switches. I ended up buying a opto shielded relay board and my limit switches are now using 12v instead of 5v and I don't have any problems anymore. But this should be the second step for you once you have a stable connection to the controller.

**Reply by spannerman52** [reply]
Thank you so much for the reply. I think that looks like my next step then? I intend to remove the router motor from the machine altogether and dry run an easel program with a pencil instead and see if it completes the task. From what I've found out the power supply that comes with this is poor at best and very noisy(electrically). No transformer just a big capacitor so perhaps it's the position of that in relation to the CNC power supply? Either way thanks for the help and I will include that in my experiments and let you know how I do.👍🏼

**Reply by spannerman52** [reply]
I'm still having problems with my cnc and it looks like I'll need an opt-isolator/coupler board fitting for the micro switches? The trouble is I know nothing very much about them. I'm using an Arduino board and my stepper motors and controllers are powered by a 24v supply. I've seen a board on Amazon that comes in the following options 12v-3.3v.  24v-3.3v.  24v-5v. 
3.3v-24v and 5v to 24v. Any suggestions as to which would be most suitable? Thanks 
![Screen Shot 2021-06-29 at 09 01 33](https://user-images.githubusercontent.com/17147691/123764230-80b92280-d8bc-11eb-8a32-6a63f6e28aa8.jpg)

**Comment by chrisla23** [comment]
I have had good luck with these on some homemade CNC machines:

https://www.tindie.com/products/eccentricwkshp/cnc-optical-limit-switch-isolator-grbl/

I found the steppers were a bigger interference than the router for me.
Good cable routing can help with that. I've also seen enough shields bonded
at both ends that it is worth mentioning that it should only be one end in
most situations.

Good luck chasing your Gremlins.

-Chris

**Reply by spannerman52** [reply]
Thanks for taking the time to help out, I have just built a separate box lined out in foil to house the power supply and plan to mount this away from the control box I have used screened cable to all the motors and switches and as yet I haven't grounded either end of the screening mesh but I plan to in the controller housing. I'll let you know how I do and thanks again
Best regards

**Comment by chrisla23** [comment]
This looks like it is also a level shifter, I would assume you don't need
that. So one that is 1:1. If you are on a standard arduino that is going to
be 5V:5V, other boards would be 3.3V:3.3V. The one on tindie I previously
sent works. I also have a 24V supply (24V tool battery connected to a 3D
printed adapter). I drive the optocoupler isolated supply side with a
separate 24V-5V DC-DC converter.

-Chris

**Reply by spannerman52** [reply]
Thank you Chris I did look at the link you sent me but sadly they do not ship to the UK that's why I was looking on Amazon. It's a problem i've battled with for years, you see something really good on an American site and it's never available in the UK..... unless it's a stone axe or an Antler pick...... one day we'll catch up🤣😂. I'll see if there are other suppliers of the same thing? 
thanks again I'll let you know how I do
Best regards
